### PR TITLE
Lexus1

### PR DESCRIPTION
**Pull request by Z1rael888**
Email Service Review!

### CODE REVIEW

File: Infrastructure/Services/EmailService.cs [hunk 1]
```diff
@@ -10,25 +10,28 @@ namespace Infrastructure.Services;
 public class EmailService : IEmailService
 {
     private readonly ISendGridClient _client;
-    private readonly string _sendGridApiKey;
     private readonly string _sendGridSenderEmail;
```
Comment (dimanoit) [inline Infrastructure/Services/EmailService.cs:13]: Do not pass all the properties separately try to use IOption pattern 
https://andrewlock.net/how-to-use-the-ioptions-pattern-for-configuration-in-asp-net-core-rc2/

**Review by dimanoit — CHANGES_REQUESTED** [review]

File: Api/Controllers/AuthController.cs [hunk 2]
```diff
@@ -27,20 +27,24 @@ public class AuthController : ControllerBase
...
+    private readonly string _confirmationTemplateId;
+    private readonly string _resetTemplateId;
 
     public AuthController(
         UserManager<Account> userManager,
         IAccountService accountService,
         IOptions<GoogleOptions> applicationSettings,
         AuthService authService,
-        IEmailService emailService, IConfiguration configuration)
+        IEmailService emailService, IConfiguration configuration, IOptions<SendGridSettings> sendGridSettings)
```
Comment (dimanoit) [inline Api/Controllers/AuthController.cs:38]: We don't need it this
Where do u use this value?

File: Api/Controllers/AuthController.cs [hunk 3]
```diff
@@ -27,20 +27,24 @@ public class AuthController : ControllerBase
...
     {
         _userManager = userManager;
         _accountService = accountService;
         _authService = authService;
         _emailService = emailService;
         _applicationSettings = applicationSettings.Value;
-        _sendGridSenderEmail = configuration["SendGrid:SenderEmail"];
+        _sendGridSenderEmail = sendGridSettings.Value.SenderEmail;
```
Comment (dimanoit) [inline Api/Controllers/AuthController.cs:45]: _sendGridSenderEmail = sendGridSettings.Value.SenderEmail;
        _confirmationTemplateId = sendGridSettings.Value.ConfirmationTemplateId;
        _resetTemplateId = sendGridSettings.Value.ResetTemplateId;
        |
        U can remove this lines

File: Api/Controllers/AuthController.cs (hunk 2 above)
Comment (Z1rael888) [reply Api/Controllers/AuthController.cs:38]: U r right,we dont use it anymore in the controller

**Pull request merged**